accepted/future-releases/extension-types/feature-specification.md

```diff
@@ -0,0 +1,1131 @@
+be a generic class, extension type, or mixin, or typedef, or function*).
+Let <code>BB<sub>j</sub></code> be the extension type erasure of
+<code>B<sub>j</sub></code>, for _j_ in _1 .. s_.
+It is a compile-time error if
+<code>X<sub>1</sub> extends BB<sub>1</sub>, .. X<sub>s</sub> extends BB<sub>s</sub></code>
+has any compile-time errors.
+
+*For example, the extension type erasure could map
+<code>X extends C<Y>, Y extends X</code> to
+<code>X extends Y, Y extends X</code>,
+which is an error.*
+
+An extension type declaration _DV_ named `Name` may declare one or
+more constructors. A constructor which is declared in an extension type
+declaration is also known as an _extension type constructor_.
+
+*The purpose of having an extension type constructor is that it bundles
+an approach for building an instance of the representation type of an
+extension type declaration _DV_ with _DV_ itself, which makes it easy to
+recognize that this is a way to obtain a value of that extension type. It
+can also be used to verify that an existing object (provided as an
+actual argument to the constructor) satisfies the requirements for
+having that extension type.*
+
+The `<representationDeclaration>` works as a constructor. The optional
+`('.' <identifier>)` in the grammar is used to declare this constructor
+with a name of the form `<identifier> '.' <identifier>` *(at times
+described as a "named constructor")*. It is a constant constructor: If `e`
+is a constant expression and `V(e)` is not an error, then `V(e)` is a
+constant expression. Other constructors may be declared `const` or not,
+following the normal rules for constant constructors.
+
+A compile-time error occurs if an extension type constructor includes a
+superinitializer. *That is, a term of the form `super(...)` or
+`super.id(...)` as the last element of the initializer list.*
+
+A compile-time error occurs if an extension type constructor declares a
+super parameter. *For instance, `Name(super.x);`.*
+
+*In the body of a generative extension type constructor, the static type of
+`this` is the same as it is in any instance member of the extension type
+declaration, that is, `Name<X1 .. Xk>`, where `X1 .. Xk` are the type
+parameters declared by `Name`.*
+
+An instance creation expression of the form
+<code>Name&lt;T<sub>1</sub>, .. T<sub>s</sub>&gt;(...)</code>
+or
+<code>Name&lt;T<sub>1</sub>, .. T<sub>s</sub>&gt;.name(...)</code>
+is used to invoke these constructors, and the type of such an expression is
+<code>Name&lt;T<sub>1</sub>, .. T<sub>s</sub>&gt;</code>.
+
+*In short, extension type constructors appear to be very similar to
+constructors in classes, and they correspond to the situation
+where the enclosing class has a single, non-late, final instance
+variable, which is initialized according to the normal rules for
+constructors (in particular, it can occur by means of `this.id`, or in
+an initializer list, or by an initializing expression in the
```

Comment:
"initializing expression" - can it?
The grammar says just `'(' <type> <identifier> ')'` - there is no place for the initializing expression.

Reply:
Good catch, thanks! So we don't have that initializing expression any more. Adjusted the text.